crates/astria-composer/src/executor/mod.rs

```diff
@@ -417,6 +431,35 @@ impl Executor {
 
         reason.map(|_| ())
     }
+
+    // check for mismatched configured chain_id and sequencer client chain_id
+    pub(crate) async fn get_sequencer_chain_id(&self) -> eyre::Result<String> {
+        let retry_config = tryhard::RetryFutureConfig::new(u32::MAX)
+            .exponential_backoff(Duration::from_millis(100))
+            .max_delay(Duration::from_secs(20))
+            .on_retry(
+                |attempt: u32,
+                 next_delay: Option<Duration>,
+                 error: &sequencer_client::tendermint_rpc::Error| {
+                    let wait_duration = next_delay
+                        .map(humantime::format_duration)
+                        .map(tracing::field::display);
+                    warn!(
+                        attempt,
+                        wait_duration,
+                        error = error as &dyn std::error::Error,
+                        "attempt to fetch sequencer genesis info; retrying after backoff",
+                    );
+                    futures::future::ready(())
+                },
+            );
+        let client_genesis: tendermint::Genesis =
+            tryhard::retry_fn(|| self.sequencer_client.genesis())
+                .with_config(retry_config)
+                .await
+                .wrap_err("failed to retrieve sequencer genesis after many attempts")?;
+        Ok(client_genesis.chain_id.to_string())
```

Comment:
The conversion is unnecessary, just return `genesis.chain_id`

Reply:
`chain_id` is of type `tendermint::chain::Id`. I may be mistaken, but I believe we'll have to de a conversion at some point for the comparison. Would you prefer to return the raw chain ID and do the conversion in `setup()`?

Reply:
You don't need to convert at all. It provides this: https://docs.rs/tendermint/latest/tendermint/chain/id/struct.Id.html#method.as_str

Reply:
Apologies for the confusion, I saw your suggestion to compare using `as_str()` below and then deleted my previous comment. Should be resolved in 8523e04 :)